WordPress/openverse
# Upgrade ESLint and Prettier packages in monorepo root

## Fixes



Fixes #2571 by @sarayourfriend 

## Description


This PR does the following:

1. Upgrade all ESLint and prettier root packages to their latest versions. I used `pnpm up --latest '*eslint*'` and `pnpm up --latest '*prettier*'` to make this easy. Originally I was going to just update the eslint plugins but it made sense to go ahead and update prettier as well. I did not update the TypeScript related ones because it's a major version jump for TS and IIRC there are some lingering issues we need to address. I'll investigate and open issues if necessary.
2. Re-enable and address the `playwright/expect-expect` warnings.
3. Fix one TS issue that required a one-line change to fix (opted for that rather than to open an issue). Filed #2907 for another TS issue.
4. Disable a new vue-a11y rule and opened an issue to investigate whether any of those need to be re-enabled. The disabled rule as a reference to the issue on each disable comment. #2906
5. Update prettier configuration to address https://github.com/tailwindlabs/prettier-plugin-tailwindcss/issues/208.


## Testing Instructions

Checkout the branch, run `pnpm install` to get the new dependency versions installed, and add some code that would cause ESLint to fail, confirm that it does. Read through the new comments etc and confirm they are accurate.

## Checklist


- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
